WalletWasabi.Fluent/ViewModels/AddWallet/Create/ConfirmRecoveryWordsViewModel.cs

```diff
@@ -54,17 +58,48 @@ public partial class ConfirmRecoveryWordsViewModel : RoutableViewModel
 		_confirmationWordsSourceList
 			.Connect()
 			.ObserveOn(RxApp.MainThreadScheduler)
-			.OnItemAdded(x => x.Reset())
 			.Sort(SortExpressionComparer<RecoveryWordViewModel>.Ascending(x => x.Index))
+			.OnItemAdded(x => x.Reset())
 			.Bind(out _confirmationWords)
 			.Subscribe();
 
-		// Select random words to confirm.
-		_confirmationWordsSourceList.AddRange(mnemonicWords.OrderBy(_ => Random.Shared.NextDouble()).Take(3));
+		_confirmationWordsSourceList.AddRange(mnemonicWords);
+
+		AvailableWords = mnemonicWords.OrderBy(_ => Random.Shared.Next()).ToList();
 	}
 
 	public ReadOnlyObservableCollection<RecoveryWordViewModel> ConfirmationWords => _confirmationWords;
 
+	public List<RecoveryWordViewModel> AvailableWords { get; }
+
+	private void OnWordSelected(RecoveryWordViewModel selectedWord, bool isSelected)
+	{
+		if (isSelected)
+		{
+			var empty = ConfirmationWords.FirstOrDefault(x => x.SelectedWord is null);
+
+			if (empty is { })
+			{
+				empty.SelectedWord = selectedWord.Word;
+			}
+		}
+		else
+		{
+			var toRemove = ConfirmationWords.FirstOrDefault(x => x.SelectedWord == selectedWord.Word);
+
+			if (toRemove is { })
+			{
+				for (int i = ConfirmationWords.IndexOf(toRemove); i < ConfirmationWords.Count; i++)
+				{
+					ConfirmationWords[i].SelectedWord =
+						i < ConfirmationWords.Count - 1
+						? ConfirmationWords[i + 1].SelectedWord
+						: null;
+				}
+			}
+		}
```

Comment:
This whole logic could be simplified by improving the UX of the selection.

I think we shouldn't allow further selection once an incorrect one was selected. We could even fade out all the items and only leave the incorrect one as it is, like marking for the user that this is incorrect, click again to remove it.

And also once an item is selected in the correct place, I wouldn't allow clicking on it again.

This method is way slower than the previous one but at least more secure. So we should apply the changes above to mitigate the possible mistakes.
I was so focused on clicking items as I didn't even realize I made a mistake, I only noticed it when I finished.